handlers/line/line.go

```diff
@@ -1,3 +1,178 @@
+	lineRequest := &moMsg{}
+	err := handlers.DecodeAndValidateJSON(lineRequest, r)
+	if err != nil {
+		return nil, courier.WriteAndLogRequestError(ctx, w, r, channel, err)
+	}
+
+	msgs := []courier.Msg{}
+	for _, lineEvent := range lineRequest.Events {
+		if (lineEvent.Source.Type == "" && lineEvent.Source.UserID == "") || (lineEvent.Message.Type == "" && lineEvent.Message.ID == "" && lineEvent.Message.Text == "") || lineEvent.Message.Type != "text" {
+
+			continue
+		}
+
+		// create our date from the timestamp (they give us millis, arg is nanos)
+		date := time.Unix(0, lineEvent.Timestamp*1000000).UTC()
+
+		urn := urns.NewURNFromParts(urns.LineScheme, lineEvent.Source.UserID, "")
+
+		msg := h.Backend().NewIncomingMsg(channel, urn, lineEvent.Message.Text).WithReceivedOn(date)
+
+		// and write it
+		err = h.Backend().WriteMsg(ctx, msg)
+		if err != nil {
+			return nil, err
+		}
+		msgs = append(msgs, msg)
+	}
+
+	if len(msgs) == 0 {
+		if len(lineRequest.Events) > 0 {
+			return nil, courier.WriteAndLogRequestIgnored(ctx, w, r, channel, "ignoring request, no message")
+		}
+		return nil, courier.WriteAndLogRequestError(ctx, w, r, channel, fmt.Errorf("missing message, source or type in the event"))
+
+	}
+
+	return []courier.Event{msgs[0]}, courier.WriteMsgSuccess(ctx, w, r, msgs)
+
+}
+
+type mtMsg struct {
+	Type string `json:"type"`
+	Text string `json:"text"`
+}
+
+type mtEnvelop struct {
+	To       string  `json:"to"`
+	Messages []mtMsg `json:"messages"`
+}
+
+// SendMsg sends the passed in message, returning any error
+func (h *handler) SendMsg(ctx context.Context, msg courier.Msg) (courier.MsgStatus, error) {
+	authToken := msg.Channel().StringConfigForKey(courier.ConfigAuthToken, "")
+	if authToken == "" {
+		return nil, fmt.Errorf("no auth token set for LN channel: %s", msg.Channel().UUID())
+	}
+
+	status := h.Backend().NewMsgStatusForID(msg.Channel(), msg.ID(), courier.MsgErrored)
+	parts := handlers.SplitMsg(handlers.GetTextAndAttachments(msg), maxMsgLength)
+	for _, part := range parts {
+		lineEnvelop := mtEnvelop{
+			To: msg.URN().Path(),
+			Messages: []mtMsg{
+				mtMsg{
+					Type: "text",
+					Text: part,
+				},
+			},
+		}
+
+		requestBody := &bytes.Buffer{}
+		json.NewEncoder(requestBody).Encode(lineEnvelop)
+
+		// build our request
+		req, err := http.NewRequest(http.MethodPost, sendURL, requestBody)
+		req.Header.Set("Content-Type", "application/json")
+		req.Header.Set("Accept", "application/json")
+		req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", authToken))
+		if err != nil {
+			return nil, err
+		}
+
+		rr, err := utils.MakeHTTPRequest(req)
+		// record our status and log
+		log := courier.NewChannelLogFromRR("Message Sent", msg.Channel(), msg.ID(), rr).WithError("Message Send Error", err)
+		status.AddLog(log)
+
+		if err != nil {
+			return status, err
+		}
+		status.SetStatus(courier.MsgWired)
```

Comment:
Any external id to parse?

Reply:
the docs says they just return a 200 and and empty JSON